docs/pages/guides/upload.mdx

```diff
@@ -0,0 +1,85 @@
+	configuration variable to `true`.
+</Callout>
+
+## Enabling upload
+
+Uploading is **disabled by default**, and is gated behind the `enable_upload` configuration variable (`true` enables uploading). This variable can be set by modifying your `Stump.toml` file to set `enable_upload: true` or by setting the environment variable `STUMP_ENABLE_UPLOAD = true`.
+
+When uploading is not enabled, the routes for uploading will not be attached to your server, and the web UI will not show the upload interface. Attempting to access uploading directly through the API will result in HTTP Error 404.
+
+### Permissions
+
+Only users with permission to upload are allowed to add files to the server. Users will need the following permissions:
+
+- `file:upload`
+- `manage:library`
+
+The server owner can assign these permissions to users on the user management settings page. For more information on that process, see the [user permissions](/guides/access-control/permissions) guide.
+
+### Configuration
+
+The server limits the maximum upload size based on your configuration, you can change `max_file_upload_size` in `Stump.toml` or the `STUMP_MAX_FILE_UPLOAD_SIZE` environment variable to set the maximum allowed upload size in bytes.
+
+## How it works
+
+This feature allows users to upload books or an entire collection of files and directories via series upload.
+
+After an upload finishes, a library scan will automatically be started to add the newly uploaded books.
+
+### Uploading books
+
+Users can upload one or more book files (e.g., `.cbz`, `.cbr`, `.epub`, and `.pdf` files) to a specific library directory.
+
+### Uploading Series
+
+Users can upload a `.zip` file containing multiple books in an arbitrary directory structure. The archive will be extracted to the location specified when uploading.
+
+### Security considerations
+
+Stump attempts to validate uploads in order to protect the host system from misuse of upload functionality, but it should be used with care. The following section describes the validation Stump applies to incoming upload requests.
+
+#### Validations
+
+- File type: Only certain file types are permitted in uploads.
+  - Books: Only `.cbr`, `.cbz`, `.epub`, and `.pdf` files are allowed.
+  - Series: Only `.zip` archives can be uploaded.
+    - The contents of the archive must also be limited to: `.cbr`, `.cbz`, `.epub`, `.pdf`, `.xml`, `.json` and image file types.
+- Content-Type: The server attempts to validate content-type of uploads to reject anything not matching the files above.
+- Path validation: The server will not allow uploads outside the library for which they are intended, and special path components like `..` will be rejected entirely.
+- Overwrite protection: An upload cannot overwrite an existing file.
+
+It may be possible to construct input that circumvents this validation. Server owners are encouraged to only enable this functionality if they need it and to only allow trusted users permission to upload files.
```

Comment:
This is important enough to be its own section (i.e. `## Security Considerations`, etc)

Reply:
Great point. I also moved this to the very bottom of the guide - not because I think it should be de-emphasized but because I think people pay the most attention to the first and last item in a document like this.

You have a better sense of user expectations, though, so tell me if you disagree and we can move it toward the top.

Reply:
No what you suggest makes total sense, I agree the “guide” part should be higher up! I don’t think it de-emphasizes it, either, at least not as a main heading